File: typescript/timer/timer.ts
export class Timer {
    private start: number;
    private stop: number;

    constructor() {
        this.start = 0;
        this.stop = 0;
    }

    Start() {
        this.start = new Date().getTime();
    }

    Stop() {
        this.stop = new Date().getTime();
    }

    Print() {
        const elapsed = this.stop - this.start;
        console.log("Execution time: " + elapsed + "ms");
    }
}

File: go/util/timer.go
package util

import (
	"fmt"
	"time"
)

type Timer struct {
	start time.Time
	stop  time.Time
}

func (t *Timer) Start() {
	t.start = time.Now()
}

func (t *Timer) Stop() {
	t.stop = time.Now()
}

func (t *Timer) Print() {
	elapsed := t.stop.Sub(t.start)
	fmt.Println("Execution time: ", elapsed)
}
